docs/decisions/0004-explicitly-declare-enterprise-for-role-assignments.rst

```diff
@@ -0,0 +1,74 @@
+Enterprise role assignments will record a customer UUID
+-------------------------------------------------------
+
+Status
+======
+
+Accepted - January 2021
+
+Context
+=======
+
+The ``SystemWideEnterpriseUserRoleAssignment`` model is used to grant role-based permissions
+to enterprise users.  These permissions generally control the users' access to data
+and ability to perform certain functions within an enterprise customer.  This assignment model does `not` currently
+refer to any specific customer - the "context" to which a role is applied is inferred from
+a user (an ``auth.User`` record) having been "linked" to a customer via the ``EnterpriseCustomerUser`` model.
+
+However, a single ``auth.User`` may be linked to multiple ``EnterpriseCustomers``, and if that user
+is assigned a role (such as ``enterprise_admin``), it is inferred that the role assignment applies
+to **every** customer to which the user is linked.  This is undesirable; we generally do not want an admin
+of `Customer A` to also have admin permissions for `Customer B` (to which the user may be linked as a mere ``learner``).
+Instead, the ``SystemWideEnterpriseUserRoleAssignment`` model should explicitly record *which* customer
```

Comment:
commentary only: TBH this name confuses me, I thought `system wide` meant actually you have access to all enterprises, but looks like I was inferring too much! But love this idea, it seems important to say the least.